SimCalorimetry/HGCalSimProducers/src/HGCDigitizerBase.cc

```diff
@@ -58,50 +73,60 @@ void HGCDigitizerBase<DFr>::runSimple(std::unique_ptr<HGCDigitizerBase::DColl>&
                                       const std::unordered_set<DetId>& validIds,
                                       CLHEP::HepRandomEngine* engine) {
   HGCSimHitData chargeColl, toa;
-
+  //const int sampleSize = (int) hgc_digi::nSamples;
   // this represents a cell with no signal charge
   HGCCellInfo zeroData;
   zeroData.hit_info[0].fill(0.f);  //accumulated energy
   zeroData.hit_info[1].fill(0.f);  //time-of-flight
+  const auto NoiseArrayBegin_ = GaussianNoiseVec_.begin();
 
-  for (const auto& id : validIds) {
+  for (std::unordered_set<DetId>::const_iterator itr = validIds.begin(); itr != validIds.end(); ++itr) {
+    int id = *itr;
     chargeColl.fill(0.f);
     toa.fill(0.f);
     HGCSimHitDataAccumulator::iterator it = simData.find(id);
     HGCCellInfo& cell = (simData.end() == it ? zeroData : it->second);
     addCellMetadata(cell, theGeom, id);
+    long hash_index((long)(id));
+    long flatRandom = CLHEP::RandFlat::shootInt(engine, (NoiseArrayLength_ - 1));
+    long hash1 = std::fabs(hash_index + flatRandom);
+    long hash2 = hash1 % NoiseArrayLength_;
+    const auto NoisePointer = NoiseArrayBegin_ + hash2;
```

Comment:
might be slightly clearer to use a reference:
`const auto& cellNoiseArray = NoiseArray[hash2];`

then in the loop below: `float randNum = cellNoiseArray[i];`